server/util/errors.ts

```diff
@@ -0,0 +1,20 @@
+export class ServerError extends Error {
+  status: number;
+  // Optionally include the original error that was thrown
+  error: any;
+  constructor(message: string, error?: any) {
+    super(message);
+    this.status = 500;
+    this.name = 'ServerError';
+    this.error = error;
+  }
+}
+
+export class AppError extends Error {
+  status: number;
+  constructor(message: string) {
```

Comment:
We may want to support taking an optional code rather than always use 400, in case we want to return a more specific type of error (e.g. 403 forbidden).

Reply:
403 errors are handled by Express middleware, so thankfully no need to catch those; and I added a new condition for 404 errors. That handles the vast majority of probable 4XX error codes. 

So I think it's a simpler developer experience to just have to think about throwing either `AppError` or `ServerError`, as the other common errors are taken care of by Express, and finally the team is notified via Rollbar about anything else that slips through